CDCgov/prime-simplereport
# Fix status in upload list table

# FRONTEND PULL REQUEST

## Related Issue

Resolves #3874 

## Changes Proposed

- change status on the upload submission list page to reflect the http status of the submission instead of the overall status, to achieve parity with the behavior of the list view page in ReportStream

## Additional Information

- originally we thought we'd need to retrieve updated overall status (I am going to call this delivery status) of a submission from the RS API to keep our data in sync with RS, but turns out the status column on the list view page reflects the http status in the response (I've been calling this submission status), which just tells us whether the file was successfully submitted or not and presumably does not change
- RS's list view page filters out upload submissions with httpStatus != 200 or 201, we only save responses to our table if the overallStatus in the response != "Error", and the overallStatus = "Error" if httpStatus != 200 or 201. therefore getting all responses from our table for an org should achieve the same behavior as in the current RS list view
    - assumption is overallStatus = "Error" iff httpStatus != 200 or 201 based on [this thread](https://skylight-hq.slack.com/archives/C03A23UMGTS/p1655835850384329)

## Testing

- Check that upload submissions list view only has Status of "Success"
- Upload a valid file using the uploader and verify it appears in the list view with status immediately showing as "Success" instead of "Pending"
 
## Screenshots / Demos

![capture](https://user-images.githubusercontent.com/6401323/176224406-aaf14080-d5c3-4436-a8c7-0085909a611e.png)


## Checklist for Author and Reviewer

### Design

### Content

### Support

### Security
